### PR TITLE
feat: improve submit application performance (FPLA-1762)

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
https://tools.hmcts.net/jira/browse/FPLA-1762

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/events/CallbackEvent.java [hunk 1]
```diff
@@ -15,6 +17,12 @@
         this.userId = requestData.userId();
     }
 
+    CallbackEvent(CallbackEvent callbackEvent) {
```
Comment (Gogetter) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/events/CallbackEvent.java:20]: Interesting approach👍

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java [hunk 2]
```diff
@@ -23,4 +27,16 @@ public static void removeTemporaryFields(CaseDetails caseDetails, String... fiel
             caseDetails.getData().remove(field);
         }
     }
+
+    public static boolean isInOpenState(CaseDetails caseDetails) {
```
Comment (Gogetter) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java:31]: just a thought, could we pass in `String` or `State`? This is as we are currently only using one attribute in `CaseDetails`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/SubmittedCaseEventHandlerTest.java [hunk 3]
```diff
@@ -1,147 +1,249 @@
...
 
-        given(cafcassEmailContentProvider.buildCafcassSubmissionNotification(callbackRequest().getCaseDetails(),
-            LOCAL_AUTHORITY_CODE)).willReturn(expectedTemplate);
+        final CaseData caseData = CaseData.builder().build();
 
-        submittedCaseEventHandler.sendEmailToCafcass(
-            new SubmittedCaseEvent(callbackRequest(), requestData));
+        @BeforeEach
+        void init() {
+            when(mapper.convertValue(any(), eq(CaseData.class))).thenReturn(caseData);
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/SubmittedCaseEventHandlerTest.java:135]: :+1

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java [hunk 4]
```diff
@@ -16,21 +20,37 @@ private CoreCaseDataStoreLoader() {
...
+        return callbackRequest(emptyMap());
+    }
+
+    public static CallbackRequest callbackRequest(Map<String, Object> placeholders) {
+        String file = readFile("core-case-data-store-api/callback-request.json", placeholders);
+        return convert(file, CallbackRequest.class);
+    }
+
+    private static String readFile(String file, Map<String, Object> placeholders) {
+        return StringSubstitutor.replace(ResourceReader.readString(file), placeholders);
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java:50]: 👏

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java (hunk 2 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java:31]: dont really agree, class name is CaseDetailsHelper, so it implies that it operates on CaseDetials. What client want to know is if case is in openedState, not if state is opened

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/SubmittedCaseEventHandlerTest.java [hunk 5]
```diff
@@ -1,147 +1,249 @@
...
 import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.LOCAL_AUTHORITY_CODE;
-import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.LOCAL_AUTHORITY_NAME;
-import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.appendSendToCtscOnCallback;
-import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.callbackRequest;
+import static uk.gov.hmcts.reform.fpl.utils.assertions.AnnotationAssertion.assertClass;
 
 @ExtendWith(SpringExtension.class)
-@SpringBootTest(classes = {SubmittedCaseEventHandler.class, JacksonAutoConfiguration.class, LookupTestConfig.class,
-    HmctsAdminNotificationHandler.class})
 public class SubmittedCaseEventHandlerTest {
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/SubmittedCaseEventHandlerTest.java:51]: not related to your changes, but we can remove the `public`

Comment (tomaszpowroznik) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/SubmittedCaseEventHandlerTest.java]: done

**Pull request merged**